rest/rest-sqs-testing-amazon-java-sdk/src/test/scala/org/elasticmq/rest/sqs/AmazonJavaSdkTestSuite.scala

```diff
@@ -950,34 +951,34 @@ class AmazonJavaSdkTestSuite extends FunSuite with Matchers with BeforeAndAfter
   }
 
   test("should create queue with redrive policy.") {
+    import spray.json._
+    import org.elasticmq.rest.sqs.model.RedrivePolicyJson._
+
     // Given
-    client.createQueue(new CreateQueueRequest("dlq1"))
+    val deadLetterQueueUrl = client.createQueue(new CreateQueueRequest("dlq1")).getQueueUrl
+    val deadLetterQueueAttributes = client.getQueueAttributes(deadLetterQueueUrl, List("All")).getAttributes
+
+    // When
+    val redrivePolicy = RedrivePolicy("dlq1", 1).toJson.toString()
+    val createQueueResult = client.createQueue(new CreateQueueRequest("q1")
+      .withAttributes(Map(defaultVisibilityTimeoutAttribute -> "1", redrivePolicyAttribute -> redrivePolicy)))
+    val newQueueAttributes = client.getQueueAttributes(createQueueResult.getQueueUrl, List("All")).getAttributes
 
     // Then
-    client.createQueue(new CreateQueueRequest("q1")
-      .withAttributes(Map(
-        defaultVisibilityTimeoutAttribute -> "1",
-        redrivePolicyAttribute ->
-          """
-            |{
-            |  "deadLetterTargetArn":"arn:aws:sqs:elasticmq:000000000000:dlq1",
-            |  "maxReceiveCount":"1"
-            |}
-          """.stripMargin
-      )))
+    deadLetterQueueAttributes.keySet() should not contain redrivePolicyAttribute
+    newQueueAttributes.keySet() should contain(redrivePolicyAttribute)
+    newQueueAttributes(redrivePolicyAttribute) should be(redrivePolicy)
   }
 
   test("should return an error when the deadletter queue does not exist") {
+    import spray.json._
+    import org.elasticmq.rest.sqs.model.RedrivePolicyJson._
+
     // Given no dead letter queue
     // Then
     a[QueueDoesNotExistException] shouldBe thrownBy {
-      client.createQueue(new CreateQueueRequest("q1").withAttributes(Map(redrivePolicyAttribute ->
-          """
-            |{
-            |  "deadLetterTargetArn":"arn:aws:sqs:elasticmq:000000000000:queueDoesNotExist",
-            |  "maxReceiveCount":"1"
-            |}
-          """.stripMargin
+      client.createQueue(new CreateQueueRequest("q1").withAttributes(Map(
+        redrivePolicyAttribute -> RedrivePolicy("queueDoesNotExist", 1).toJson.toString()
```

Comment:
Maybe a test that the `RedrivePolicy` converts to the expected json would be good to have as well? Since we kind of tested this here - giving the raw json to the request, and now  I think there's no place where the to-json conversion is checked

Reply:
I've expanded the tests a bit in `RedrivePolicyJsonTest` for verifying (de)serializing of `RedrivePolicy` instances.